src/App.tsx

```diff
@@ -1,54 +1,90 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [choosenPerson, setChoosenPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setApliedQuery] = useState('');
+  const [isImputFocus, setIsImputFocus] = useState(false);
+
+  const applyQwery = useCallback(
+    debounce(setApliedQuery, 1000),
+    [],
+  );
+
+  const handleQweryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQwery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person => (
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase())
+    ));
+  }, [appliedQuery, peopleFromServer]);
+
+  const handlKey = (
```

Comment:
The function name 'handlKey' is not very descriptive and has a typo, it should be 'handleKey'. Always provide descriptive names for your functions to improve code readability.